src/lib.rs

```diff
@@ -74,6 +98,29 @@ impl Neo4j {
         )
     }
 
+    /// Define Neo4j lab plugins to get started with the database.
+    /// Returns new instance.
+    #[must_use]
+    pub fn with_neo4j_labs_plugin(&self, plugins: Vec<Neo4jLabsPlugin>) -> Self {
+        let mut env_vars = self.env_vars.clone();
+
+        let plugin_names = plugins.into_iter()
+            .map(|p| format!("\"{}\"", p.get_value()))
+            .collect::<Vec<String>>()
+            .join(",");
+
+        let plugin_definition = format!("[{}]", plugin_names);
+
+        env_vars.insert("NEO4JLABS_PLUGINS".to_owned(), plugin_definition);
+
+        Self {
+            version: self.version.clone(),
+            user: self.user.clone(),
+            pass: self.pass.clone(),
+            env_vars,
+        }
+    }
```

Comment:
These builder-style/with APIs usually take ownership of `self`, this removes the question of what happens to the value that you had before and doesn't require you to clone all the things.

```suggestion
    pub fn with_neo4j_labs_plugin(mut self, plugins: Vec<Neo4jLabsPlugin>) -> Self {
        let plugin_names = plugins.into_iter()
            .map(|p| format!("\"{}\"", p)) // can be just `p` if we implement `Display` as above
            .collect::<Vec<String>>()
            .join(",");

        let plugin_definition = format!("[{}]", plugin_names);

        self.env_vars.insert("NEO4JLABS_PLUGINS".to_owned(), plugin_definition);

        self
    }
```

Doing this also means we don't have to add `#[must_use]`, as there is no way to continue without using.

Reply:
Also, since you only need to read from `plugins` and don't actually need it to be a `Vec`, it is more idiomatic to accept a slice instead

```suggestion
    pub fn with_neo4j_labs_plugin(mut self, plugins: &[Neo4jLabsPlugin]) -> Self {
        let plugin_names = plugins.iter()
            .map(|p| format!("\"{}\"", p))
            .collect::<Vec<String>>()
            .join(",");

        let plugin_definition = format!("[{}]", plugin_names);

        self.env_vars.insert("NEO4JLABS_PLUGINS".to_owned(), plugin_definition);

        self
    }
```

Reply:
Also, does it make sense to skip empty lists? This also works well with the `self` receiver:

```suggestion
    pub fn with_neo4j_labs_plugin(mut self, plugins: &[Neo4jLabsPlugin]) -> Self {
        if plugins.is_empty() {
            return self;
        }

        let plugin_names = plugins
            .iter()
            .map(|p| format!("\"{}\"", p))
            .collect::<Vec<String>>()
            .join(",");

        let plugin_definition = format!("[{}]", plugin_names);

        self.env_vars
            .insert("NEO4JLABS_PLUGINS".to_owned(), plugin_definition);

        self
    }
```

Reply:
And lastly, I might want to accept a more generic "anything that can be iterated over" parameter.
Here probably not, it makes the signature quite complex, and it gets even weirder if you want to abstract over `Neo4jLabsPlugin` as well as `&Neo4jLabsPlugin` as the iterator item. The signature might start to look like

```rust
    pub fn with_neo4j_labs_plugin<'x, I: IntoIterator<Item = &'x Neo4jLabsPlugin>>(
        mut self,
        plugins: I,
    ) -> Self {
```

and I'm not gonna continue with that. I think the slice is the best option here, so just as a side note…